src/field.rs

```diff
@@ -190,6 +196,55 @@ pub trait FieldElement:
         }
         Ok(vec)
     }
+
+    /// Encode `input` as `bits`-bit vector of elements of `Self` if it's small enough
+    /// to be represented with that many bits.
+    ///
+    /// # Arguments
+    ///
+    /// * `input` - The field element to encode
+    /// * `bits` - The number of bits to use for the encoding
+    /// * `max_input` - The maximal number representable with `bits` bits. Equal to `bits^2-1`
+    fn encode_into_bitvector_representation(
+        input: &Self::Integer,
+        bits: usize,
+        max_input: Self::Integer,
+    ) -> Result<Vec<Self>, FieldError> {
+        if *input > max_input {
+            return Err(FieldError::InputSizeMismatch);
+        }
+
+        let one = Self::Integer::from(Self::one());
+        let mut encoded = Vec::with_capacity(bits);
+        for l in 0..bits {
+            let l = Self::Integer::try_from(l).map_err(|_| {FieldError::IntegerTryFrom})?;
+            let w = Self::from((*input >> l) & one);
+            encoded.push(w);
+        }
+        Ok(encoded)
+    }
+
+    /// Decode the bitvector-represented value `input` into a simple representation as a single field element.
+    /// This function panics if `2 ^ input.len()` does not fit into the field `Self`.
+    fn decode_from_bitvector_representation(input: &[Self]) -> Self {
```

Comment:
Now that this method is in the public API, I think it's best to err rather than panic.

Reply:
Yep, we just did this.